lib/client/conntest/ssh.go

```diff
@@ -49,30 +60,212 @@ func NewSSHConnectionTester(cd services.ConnectionsDiagnostic) *SSHConnectionTes
 //   - the SSH Node will append a trace indicating if the has access (RBAC)
 //   - the SSH Node will append a trace indicating if the requested principal is valid for the target Node
 func (s *SSHConnectionTester) TestConnection(ctx context.Context, req TestConnectionRequest) (types.ConnectionDiagnostic, error) {
-	id := uuid.NewString()
-	connectionDiagnostic, err := types.NewConnectionDiagnosticV1(id, map[string]string{},
+	connectionDiagnosticID := uuid.NewString()
+	connectionDiagnostic, err := types.NewConnectionDiagnosticV1(connectionDiagnosticID, map[string]string{},
 		types.ConnectionDiagnosticSpecV1{
-			Message: types.DiagnosticMessageWaiting,
+			Message: types.DiagnosticMessageFailed,
```

Comment:
Why do we set it to "failed" initially? Wouldn't it make sense to leave it unset?

Reply:
Removing the initial message. :+1:

Reply:
This way we can always exit the function by appending a new error trace and returning.

If we don't set it to Failed, we'll need to append the trace, update the message and call the Update method.

So, the error path would look like this:
```go
	if err != nil {
		connDiag, err := s.clt.AppendTraceConnectionDiagnostic(ctx, connectionDiagnosticID, types.NewFailedTraceConnectionDiagnostic(
			types.DiagnosticTraceTypeRBACNode,
			"Node not found. Ensure the Node exists and your role allows you to access it.",
			err,
		))
		if err != nil {
			return nil, trace.Wrap(err)
		}

		connDiag.SetMessage(types.DiagnosticMessageFailed)
		if err := s.clt.UpdateConnectionDiagnostic(ctx, connDiag); err != nil {
			return nil, trace.Wrap(err)
		}

		return connDiag, nil
	}
```

Instead of
```go
	if err != nil {
		connDiag, err := s.clt.AppendTraceConnectionDiagnostic(ctx, connectionDiagnosticID, types.NewFailedTraceConnectionDiagnostic(
			types.DiagnosticTraceTypeRBACNode,
			"Node not found. Ensure the Node exists and your role allows you to access it.",
			err,
		))
		if err != nil {
			return nil, trace.Wrap(err)
		}

		return connDiag, nil
	}
```

We do repeat this multiple times, so I think the current approach makes this clearer.

Reply:
Okay that's fair. I'd add a comment here then to explain this.